node/grpc/run.go

```diff
@@ -65,11 +57,7 @@ func RunServers(serverV1 *Server, serverV2 *ServerV2, config *node.Config, logge
 			return
 		}
 		for {
-			addr := fmt.Sprintf("%s:%s", localhost, config.InternalV2DispersalPort)
-			listener, err := net.Listen("tcp", addr)
-			if err != nil {
-				logger.Fatalf("Could not start tcp listener: %v", err)
-			}
+			listener := serverV2.dispersalListener
```

Comment:
In the PR description, you mention we are making this change in order to use OS allocated ports. Is there a reason why we can't update `config` to point to those OS allocated ports instead of adding new constructor arguments?

Reply:
Right now the way we get a port allocated by the OS is to create a listener with port 0.
```go
	v1DispersalListener, err := net.Listen("tcp", "0.0.0.0:0")
```

Not sure of the alternative for getting a random port.

Reply:
Ah, ok. Not a huge fan of passing in test-only things to the constructor, but I'm ok moving forward with this PR to unblock inabox improvements. We can work on fixing the constructor arguments in the future. (Perhaps some sort of interface responsible for building sockets, and we can pass in a mock implementation?)

Reply:
Didn't look at how this is actually implemented here, but @cody-littley the port would actually be part of the (external) config! If a user sets port 0 as part of his config, then the OS will spin up the service on a random allocated port. This is most useful for testing, but not strictly for that use case.

The alternative to doing listener injection is to let the constructor create the listener, and then expose a public function `Port()`. I personally think the former is cleaner, but open to bike-shed debating this :)